res/controllers/Traktor-Kontrol-S3-hid-scripts.js

```diff
@@ -0,0 +1,2227 @@
+        "FF FF FF FF  00 00 00 00  00 00 00 00  00 00 00 00 " +
+        "00 00 00 00  00 00 00 00  00 00 00 00  00 00 00 00 " +
+        "00 00 00 00 ",
+        "00 00 00  00 00 00 00  00 00 00 00  00 00 00 00 " +
+        "00 00 00 00  00 00 00 00  00 00 00 00  00 00 00 00 " +
+        "00 00 00 00  00 00 00 00  00 00 00 00  00 00 00 00 " +
+        "00 00 00 00  00 00 00 00  00 00 00 00  00 00 00 00 " +
+        "00 00 00 00  00 00 00 00  00 00 00 00  00 00 00",
+        "20 08 00  00 00 00 00  00 00 00 00  00 00 00 00 " +
+        "00 00 00 00  00 00 00 00  00 00 00 00  00 00 00 00 " +
+        "00"
+    ];
+
+    var data = [Array(), Array(), Array()];
+
+
+    for (var i = 0; i < data.length; i++) {
+        var ok = true;
+        var tokens = dataStrings[i].split(/\s+/);
+        HIDDebug("i " + i + " " + tokens);
+        data[i].length = tokens.length;
+        for (var j = 0; j < tokens.length; j++) {
+            var byteStr = tokens[j];
+            if (byteStr.length === 0) {
+                continue;
+            }
+            if (byteStr.length !== 2) {
+                ok = false;
+                HIDDebug("not two characters?? " + byteStr);
+            }
+            var b = parseInt(byteStr, 16);
+            if (b < 0 || b > 255) {
+                ok = false;
+                HIDDebug("number out of range: " + byteStr + " " + b);
+            }
+            data[i][j] = b;
+        }
+        if (ok) {
+            var header = 0x80 + i;
+            if (i === 2) {
+                header = 0xF4;
+            }
+            controller.send(data[i], data[i].length, header);
+        }
+    }
+    TraktorS3.kontrol.setInputLineMode(false);
+};
+
+TraktorS3.shutdown = function() {
+    // Deactivate all LEDs
+    var dataStrings = [
+        "      00 00  00 00 00 00  00 00 00 00  00 00 00 00 " +
+        "00 00 00 00  00 00 00 00  00 00 00 00  00 00 00 00 " +
+        "00 00 00 00  00 00 00 00  00 00 00 00  00 00 00 00 " +
+        "00 00 00 00  00 00 00 00  00 00 00 00  00 00 00 00 " +
+        "00 00 00 00  00 00 00 00  00 00 00 00  00 00 00 00 " +
+        "00 00 00 00 ",
+        "      00 00  00 00 00 00  00 00 00 00  00 00 00 00 " +
+        "00 00 00 00  00 00 00 00  00 00 00 00  00 00 00 00 " +
+        "00 00 00 00  00 00 00 00  00 00 00 00  00 00 00 00 " +
+        "00 00 00 00  00 00 00 00  00 00 00 00  00 00 00 00 " +
+        "00 00 00 00  00 00 00 00  00 00 00 00  00 00 00",
+    ];
+
+    var data = [Array(), Array()];
+
+
+    for (var i = 0; i < data.length; i++) {
+        var tokens = dataStrings[i].split(/\s+/);
+        data[i].length = tokens.length;
+        for (var j = 0; j < tokens.length; j++) {
+            var byteStr = tokens[j];
+            if (byteStr.length === 0) {
+                continue;
+            }
+            data[i][j] = parseInt(byteStr, 16);
+        }
+        controller.send(data[i], data[i].length, 0x80 + i);
+    }
```

Comment:
This is quite a complex way to fill an array with 0s.

Reply:
You make an excellent point, I'll fix this

Reply:
ping?

Reply:
fixed